### ISSUE TITLE
CAT08 Dataset or make own dataset?

### ISSUE THREAD

**Issue by 1hell0the1others8** [BubblyYi/Coronary-Artery-Tracking-via-3D-CNN-Classification#20]
Anyone could share CAT08 Dataset for research? The website link of this dataset is invalid now. 
Or Who shall teach me how to make my own dataset for coronary centerline tracking? 
By the way, I have no idea how to get the reference radius r based on centerlines and CTA image patch. For a centain point x, is the radius r a fixed value which means coronary vessel lumen radius or just a random value?

**Comment by 3Dlion** [comment]
same request for dataset! I have used Mimics software to make own dataset. Mimics has a semi-automatic tool of coronary extraction. Many other medical image processing software has the function of centerline extraction, but you have to segment vessels firstly.

**Comment by liyuxuan89** [comment]
Is there new link to the dataset. I also find the dataset's link invalid.

**Comment by gregpost** [comment]
I suggest to build the dataset together. I already have my own dataset with 38 CT series. But if somebody have another dataset this is good idea to combine our datasets together. You can send me an email to exchange our datasets.

**Comment by Manasi-25** [comment]
did anyone here got the access to the dataset?? If yes Please share!!

**Comment by jinaozhang** [comment]
同求数据集，有的话麻烦分享一下，万分感谢！

**Comment by francescopisu** [comment]
@gregpost do you think the [ASOCA dataset](https://asoca.grand-challenge.org/) could be integrated with CAT08 ? I didn't manage to download CAT08 but I have ASOCA. Whoever has any idea about how the CAT08 is structured, what kind of format is used, how the labels and centerlines are provided, etc. can send me a message on [LinkedIn](https://www.linkedin.com/in/francescopisu/)

**Comment by journey1234-liu** [comment]
Any updates on this issue?

**Comment by francescopisu** [comment]
@journey1234-liu No updates: CAT08 can't be shared anymore; ASOCA apparently had the same fate, unfortunately

**Comment by gregpost** [comment]
We made our own dataset. I can show you the example of this:
https://disk.yandex.ru/d/LR-C42NwDC7RRA

**Comment by gregpost** [comment]
@francescopisu @journey1234-liu
We made our own dataset. I can show you the example of this:
https://disk.yandex.ru/d/LR-C42NwDC7RRA

**Comment by francescopisu** [comment]
@gregpost BASED. Thank you. Did you have to balance positive/negative ostia and seed patches and offset/no offset centerline patches when generating the train and val datasets?

**Comment by laihanQ** [comment]
> [@francescopisu](https://github.com/francescopisu) [@journey1234-liu](https://github.com/journey1234-liu) We made our own dataset. I can show you the example of this: https://disk.yandex.ru/d/LR-C42NwDC7RRA

Hi, may i ask that could i get your own dataset for research?

**Comment by AryanMoradi** [comment]
Hey everyone, I have downloaded [ImageCAS ](https://www.kaggle.com/datasets/xiaoweixumedicalai/imagecas) dataset which has 1000 CCTAs. The issue is that it only contains image and segmentation label nifti files. And the vessel segmentations are not separated. But if anyone is down, we can work together and adjust it to match the structure of CAT08. I saw @gregpost dataset, it's great. I think we can expand it even further with ImageCAS.

**Comment by shuerwhy** [comment]
> Hey everyone, I have downloaded [ImageCAS ](https://www.kaggle.com/datasets/xiaoweixumedicalai/imagecas) dataset which has 1000 CCTAs. The issue is that it only contains image and segmentation label nifti files. And the vessel segmentations are not separated. But if anyone is down, we can work together and adjust it to match the structure of CAT08. I saw [@gregpost](https://github.com/gregpost) dataset, it's great. I think we can expand it even further with ImageCAS.

Hi,I have viewed ImageCAS dataset and another coronary artery dataset in Hugging face，but they don't have the annotation of radius either.Can you tell me how do you solve this problem?

**Comment by AryanMoradi** [comment]
@shuerwhy We could potentially use VMTK's centreline extraction. On the backend logic, it just uses a Voronoi diagram along the vessel segmentation [here](http://www.vmtk.org/tutorials/Centerlines.html) The only issue with that is that it requires the inlets and outlets of the vessels to be specified. We could use 3DSlicer automatic inlet and outlet extraction tool for that **however the issue with that is that it doesn't detect the inlets and outlets of the vessels correctly sometimes, and I haven't figured out why this happens.**
But once we have that pipeline setup, the rest should be smooth sailing, since the VMTK extracts almost perfect centrelines given the correct inlet and outlet points on the segmentation. And from the centrelines, extracting the radius should be fairly similar to the original paper for this repository. We can just calculate the largest sphere that can fit inside the vessel a any given point on the extracted centreline, and grab its radius.
Also one thing to note is that the ImageCAS data has vessel coronary segmentations as two distinct structures inside the same Nifti file. The VMKT centreline extraction assumes that it is working **only on one** vessel structure, so when it is applied to ImageCAS segmentation data, it only picks up one of the two coronaries. I used 3DSlicer **island** function to isolate each structures and work on each of them separately and once I got the centrelines for both, I could just extract that as .nrrd or even .txt file later on. That worked, but then again, the automatic inlet and outlet extraction is not perfect. And annotating inlets and outlets on 1000 segmentation files is redundant in my opinion, so surely we can find a solution to that.
Once this is done, we can use a simple script to split the extracted centrelines based on bifurcations, same way its done in @gregpost data, so we will end up with multiple centreline .txt files for a given volume.

**Comment by shuerwhy** [comment]
@AryanMoradi  Your response is so detailed,thank you so much and good luck with your experiment!

**Comment by shuerwhy** [comment]
> 我们制作了自己的数据集。我可以给你看这个例子：https://disk.yandex.ru/d/LR-C42NwDC7RRA

Hi, sorry to bother you.I have viewed your dataset  and noticed that some of the annotations don't seem to match the corresponding images. I'm not sure if I'm misunderstanding the annotation format or if there might be an issue with the labeling.

Could you please take a look when you have a moment? I'd really appreciate any clarification you can provide.

**Comment by AryanMoradi** [comment]
Has anyone managed to train a model using using @gregpost data? My centerline code seems to train just fine but the validation loss does not improve at all. But then the seedpoint and ostia training loops fail halfway through and dont improve. I'd appreciate any help. Im guessing there is something wrong with how I refactored the code to match the dataset?
